### ISSUE TITLE
generated class overrides final method

### ISSUE THREAD

**Issue by caplan** [immutables/immutables#250]
In this situation:

```java
@Value.Immutable
public abstract class Takeout extends Lunch implements Meal {
  // ...
}
```
```java 
public interface Meal { String taste(); }
```
```java 
public abstract class Lunch {
    public final String taste() {
        return "chicken";
    }
}
```

, the generated `ImmutableTakeout` class tries to override the final `Lunch.taste()` method.

**Comment by elucash** [comment]
Wow, this this surprising... and reproducible, thank you for revealing that! It just had to be fixed

**Comment by caplan** [comment]
cool - let me know if you'd like a pr for this.

**Comment by elucash** [comment]
If you already know where the wrong check is, then PR would be very welcome, if not, then I probably would have a better chance to fix it )

**Comment by caplan** [comment]
I'll let you look into it then :)

**Comment by elucash** [comment]
Released in 2.1.10. Thanks!
